sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -137,13 +137,21 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
       plan: LogicalPlan): (Option[Expression], LogicalPlan) = {
     var newPlan = plan
     val newExprs = exprs.map { e =>
-      e transformUp {
+      e transformDown {
         case Exists(sub, conditions, _) =>
           val exists = AttributeReference("exists", BooleanType, nullable = false)()
           // Deduplicate conflicting attributes if any.
           newPlan = dedupJoin(
             Join(newPlan, sub, ExistenceJoin(exists), conditions.reduceLeftOption(And)))
           exists
+        case (Not(InSubquery(values, ListQuery(sub, conditions, _, _)))) =>
+          val exists = AttributeReference("exists", BooleanType, nullable = false)()
```

Comment:
nit: There are duplicate codes in `case (Not(InSubqyer...` and `case InSubquery...`. Can we make a simple helper method to remove these?

Reply:
yea, it's ok to keep the current one. Thanks!